doc/contrib/unit_tests.rst

```diff
@@ -0,0 +1,232 @@
+
+You may try running your test by following instructions in :ref:`this section <running_pytest>`.
+
+C++: Google Test
+================
+Add your test under the directory `tests/cpp/ <https://github.com/dmlc/xgboost/tree/master/tests/cpp>`_. Refer to `this excellent tutorial on using Google Test <https://developer.ibm.com/articles/au-googletestingframework/>`_.
+
+You may try running your test by following instructions in :ref:`this section <running_gtest>`.
+
+JVM packages: JUnit / scalatest
+===============================
+The JVM packages for XGBoost (XGBoost4J / XGBoost4J-Spark) use `the Maven Standard Directory Layout <https://maven.apache.org/guides/introduction/introduction-to-the-standard-directory-layout.html>`_. Specifically, the tests for the JVM packages are located in the following locations:
+
+* `jvm-packages/xgboost4j/src/test/ <https://github.com/dmlc/xgboost/tree/master/jvm-packages/xgboost4j/src/test>`_
+* `jvm-packages/xgboost4j-spark/src/test/ <https://github.com/dmlc/xgboost/tree/master/jvm-packages/xgboost4j-spark/src/test>`_
+
+To write a test for Java code, see `JUnit 5 tutorial <https://junit.org/junit5/docs/current/user-guide/>`_.
+To write a test for Scala, see `Scalatest tutorial <http://www.scalatest.org/user_guide/writing_your_first_test>`_.
+
+You may try running your test by following instructions in :ref:`this section <running_jvm_tests>`.
+
+R package: testthat
+===================
+Add your test under the directory `R-package/tests/testthat <https://github.com/dmlc/xgboost/tree/master/R-package/tests/testthat>`_. Refer to `this excellent tutorial on testthat <https://kbroman.org/pkg_primer/pages/tests.html>`_.
+
+You may try running your test by following instructions in :ref:`this section <running_r_tests>`.
+
+**************************
+Running Unit Tests Locally
+**************************
+
+.. _running_r_tests:
+
+R package
+=========
+Run
+
+.. code-block:: bash
+
+  make Rcheck
+
+at the root of the project directory.
+
+.. _running_jvm_tests:
+
+JVM packages
+============
+As part of the building process, tests are run:
+
+.. code-block:: bash
+
+  mvn package
+
+.. _running_pytest:
+
+Python package: pytest
+======================
+
+.. note:: Having issue? Try Docker container
+
+  If you are having difficulty running the commands below (e.g. due to missing packages), consider using our Docker container. See :ref:`running_tests_inside_docker`.
+
+To run Python unit tests, first install `pytest <https://docs.pytest.org/en/latest/contents.html>`_ package:
+
+.. code:: bash
+
+  pip3 install --user pytest
+
+Then compile XGBoost:
+
+.. code:: bash
+
+  mkdir build
+  cd build
+  cmake ..
+  make
+  cd ..
+
+Now invoke pytest at the project root directory:
+
+.. code:: bash
+
+  export PYTHONPATH=./python-package
+  pytest -v -s --fulltrace tests/python
+
+In addition, to build and test CUDA code, run:
+
+.. code:: bash
+
+  cd build
+  cmake -DUSE_CUDA=ON -DUSE_NCCL=ON ..
```

Comment:
Same as above, we have build.rst.  Please try to cite that file instead of repeating it here.